### PR TITLE
Do not schedule execution of save operator if other computer operator…

### PR DESCRIPTION
**Pull request by jpurusho65**
…s fail in a DAG

## Describe your changes and why you are making these changes
Schedule the Load (Save) operator for execution only if all other compute operators have succeeded

## Related issue number (if any)
ENG-2661

## Checklist before requesting a review
- [x ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ x] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/engine/aq_engine.go [hunk 1]
```diff
@@ -1075,26 +1083,40 @@ func (eng *aqEngine) execute(
...
-						return errors.Newf("Internal error: operator %s has a negative dependnecy count.", op.Name())
+						return errors.Newf("Internal error: operator %s has a negative dependency count.", op.Name())
 					}
 
 					if opToDependencyCount[nextOp.ID()] == 0 {
 						// Defensive check: do not reschedule an already in-progress operator. This shouldn't actually
 						// matter because we only keep and update a single copy an on operator.
 						if _, ok := inProgressOps[nextOp.ID()]; !ok {
-							inProgressOps[nextOp.ID()] = nextOp
+							if nextOp.Type() != workflow_ops.LoadType {
```
Comment (kenxu95) [inline src/golang/lib/engine/aq_engine.go:1099]: Can you add a comment here about why we're skipping the load operators? And why does this check not matter on the second pass?

File: src/golang/lib/engine/aq_engine.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ import (
 	shared_utils "github.com/aqueducthq/aqueduct/lib/lib_utils"
 	"github.com/aqueducthq/aqueduct/lib/models"
 	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	workflow_ops "github.com/aqueducthq/aqueduct/lib/models/shared/operator"
```
[the remark this answers is not in the document]
Comment (jpurusho65) [reply src/golang/lib/engine/aq_engine.go]: done

File: src/golang/lib/engine/aq_engine.go (hunk 1 above)
Comment (jpurusho65) [reply src/golang/lib/engine/aq_engine.go]: done

File: src/golang/lib/engine/aq_engine.go [hunk 3]
```diff
@@ -1075,26 +1083,40 @@ func (eng *aqEngine) execute(
...
+								log.Infof("Skip load operator %s", nextOp.Name())
+							}
 						}
 					}
 				}
 			}
 
 			time.Sleep(timeConfig.OperatorPollInterval)
 		}
+		if len(inProgressOps) == 0 && !loadOpsDone {
```
[the remark this answers is not in the document]
Comment (jpurusho65) [reply src/golang/lib/engine/aq_engine.go:1118]: done

**Review by kenxu95 — CHANGES_REQUESTED** [review]
This seems to still be missing the new integration test for this?

File: src/golang/lib/engine/aq_engine.go [hunk 4]
```diff
@@ -952,12 +958,19 @@ func (eng *aqEngine) execute(
 
 	start := time.Now()
 
+	// We defer save operations until all other computer operations are completed successfully.
+	// This flag tracks whether the save operations are scheduled for execution.
+	loadOpsDone := false
+
+	// failedOps is incremented when any compute operations fail
+	failedOps := 0
```
Comment (kenxu95) [inline src/golang/lib/engine/aq_engine.go:966]: I actually don't think we need this counter. Let's chat at our 1:1

**Comment by jpurusho65** [comment]
> This seems to still be missing the new integration test for this?

done

**Review by kenxu95 — APPROVED** [review]
LGTM thanks!

**Pull request merged**